### PR TITLE
py-bx-python: add 0.9.0

### PR DESCRIPTION
**Pull request by A-N-Other**
Updating to `@0.9.0`.

Cleaning up the dependencies:
- new builds don't need `six`
- switch to `python@3.7:3.10`
- add missing `py-python-lzo`

Depends on: https://github.com/spack/spack/pull/38418 ... concretizer won't like it until that's merged.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-bx-python/package.py [hunk 1]
```diff
@@ -13,10 +13,12 @@ class PyBxPython(PythonPackage):
     homepage = "https://github.com/bxlab/bx-python"
     pypi = "bx-python/bx-python-0.8.8.tar.gz"
 
+    version("0.9.0", sha256="fe545c44d2ea74b239d41e9090618aaf6a859d1a1f64b4a21b133cb602dfdb49")
     version("0.8.8", sha256="ad0808ab19c007e8beebadc31827e0d7560ac0e935f1100fb8cc93607400bb47")
 
-    depends_on("python@2.7:2.8,3.5:", type=("build", "run"))
+    depends_on("python@3.7:3.10", when="@0.8.13:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bx-python/package.py]: Where do you see the upper bound on Python version?

File: var/spack/repos/builtin/packages/py-bx-python/package.py [hunk 2]
```diff
@@ -13,10 +13,12 @@ class PyBxPython(PythonPackage):
...
     depends_on("py-setuptools", type="build")
     depends_on("py-cython", type="build")
     depends_on("py-numpy", type=("build", "run"))
-    depends_on("py-six", type=("build", "run"))
+    depends_on("py-six", when="@:0.8.9", type=("build", "run"))
+    depends_on("py-python-lzo@1.14:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bx-python/package.py:25]: Might be useful to add a comment saying that this dep isn't listed but is still imported at run-time

File: var/spack/repos/builtin/packages/py-bx-python/package.py (hunk 1 above)
Comment (A-N-Other) [reply var/spack/repos/builtin/packages/py-bx-python/package.py]: Corrected that

File: var/spack/repos/builtin/packages/py-bx-python/package.py (hunk 2 above)
Comment (A-N-Other) [reply var/spack/repos/builtin/packages/py-bx-python/package.py:25]: Clarified that. It was hiding in the `tox.ini` file ...

**Review by adamjstewart — APPROVED** [review]

**Comment by A-N-Other** [comment]
@spackbot run pipeline

**Comment by adamjstewart** [comment]
You'll need to merge/rebase develop to get the python-lzo updates.

**Comment by A-N-Other** [comment]
Oops, didn't realise I hadn't. Thanks!

**Pull request merged**